joint_state_publisher/joint_state_publisher/joint_state_publisher

```diff
@@ -216,11 +216,18 @@ class JointStatePublisher():
                     parent = param['parent']
                     factor = param.get('factor', 1)
                     offset = param.get('offset', 0)
+                    # Handle recursive mimic chain
+                    recursive_mimic_chain_joints = [name]
                     while parent in self.dependent_joints:
+                        if parent in recursive_mimic_chain_joints:
+                            error_message = "Found an infinite recursive mimic chain"
+                            rospy.logerr("%s: [%s, %s]", error_message, ', '.join(recursive_mimic_chain_joints), parent)
+                            sys.exit(-1)
+                        recursive_mimic_chain_joints.append(parent)
                         param = self.dependent_joints[parent]
                         parent = param['parent']
                         factor *= param.get('factor', 1)
-                        # the offset is relative only to the first parent
+                        offset = factor*offset + param.get('offset', 0)
```

Comment:
Hi thank you for thew changes,

I think the correct math is:

  offset += factor * param.get('offset', 0)
  factor *= param.get('factor', 1)

(in that order)